rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -193,16 +193,19 @@ object ProcNormalizeMatcher {
             (result.chan :: acc._1, result.knownFree)
           }
         )
-        val newEnv = input.env.absorbFree(formalsResults._2)
+        val newEnv       = input.env.absorbFree(formalsResults._2)
+        val newFreeCount = formalsResults._2.next - input.knownFree.next
```

Comment:
It should just be formalsResults._2.next.
What you have may even end up being negative.

Reply:
Yeah the formalsResults starts with its own empty map; thanks for the catch.